tiledb/sm/cpp_api/query.h

```diff
@@ -1678,6 +1678,43 @@ class Query {
         name, std::get<0>(buf), std::get<1>(buf), std::get<2>(buf));
   }
 
+  /**
+   * Sets a buffer for a string-typed variable-sized, nullable attribute.
+   *
+   * @param name Attribute name
+   * @param offsets Offsets where a new element begins in the data buffer.
+   * @param data Pre-allocated string buffer.
+   * @param validity_bytemap Buffer vector with elements of the attribute
+   *     validity values.
+   **/
+  Query& set_buffer_nullable(
+      const std::string& name,
+      std::vector<uint64_t>& offsets,
+      std::string& data,
+      std::vector<uint8_t>& validity_bytemap) {
+    // Checks
+    auto is_attr = schema_.has_attribute(name);
+    if (!is_attr)
+      throw TileDBError(
+          std::string("Cannot set buffer; Attribute '") + name +
+          "' does not exist");
+    else
+      impl::type_check<char>(schema_.attribute(name).type());
+
+    // Compute element size (in bytes).
+    size_t element_size = tiledb_datatype_size(schema_.attribute(name).type());
+
+    return set_buffer_nullable(
+        name,
+        offsets.data(),
+        offsets.size(),
+        &data[0],
+        data.size(),
+        element_size,
```

Comment:
This line looks correct, but let's use `sizeof(char)` here to exactly match the implementation of non-nullable sibling: https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/cpp_api/query.h#L1435